tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,654 @@
+
+TMP_DIR = '/tmp'
+ACL_TABLE_FILE = 'acl_tbl_config.json'
+ACL_TABLE_REMOVE_FILE = "acl_tbl_del.json"
+
+ACL_RULES_FILE = 'acl_rule_config.json'
+ACL_RULES_REMOVE_FILE = "acl_rules_del.json"
+ACL_TEMPLATE_TABLE_FILE = "dscp_acl_tablev4_v6.j2"
+ACL_TEMPLATE_RULE_FILE = "dscp_acl_rulev4_v6.j2"
+
+TABLE_V4 = "OVERLAY_MARK_META_TEST"
+TABLE_V6 = "OVERLAY_MARK_META_TESTV6"
+
+
+@pytest.fixture(
+    name="encap_type",
+    scope="module",
+    params=SUPPORTED_ENCAP_TYPES)
+def fixture_encap_type(request):
+    '''
+        This fixture forces the script to perform one encap_type at a time.
+        So this script doesn't support multiple encap types at the same.
+    '''
+    return request.param
+
+
+@pytest.fixture(scope='module')
+def prepare_test_port(rand_selected_dut, tbinfo):
+    mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+    dut_port = list(mg_facts['minigraph_portchannels'].keys())[0]
+    if not dut_port:
+        pytest.skip('No portchannels found')
+    if "Ethernet" in dut_port:
+        dut_eth_port = dut_port
+    elif "PortChannel" in dut_port:
+        dut_eth_port = mg_facts["minigraph_portchannels"][dut_port]["members"][0]
+    ptf_src_port = mg_facts["minigraph_ptf_indices"][dut_eth_port]
+
+    topo = tbinfo["topo"]["type"]
+    # Get the list of upstream ports
+    upstream_ports = defaultdict(list)
+    upstream_port_ids = []
+    for interface, neighbor in list(mg_facts["minigraph_neighbors"].items()):
+        port_id = mg_facts["minigraph_ptf_indices"][interface]
+        if (topo == "t1" and "T2" in neighbor["name"]):
+            upstream_ports[neighbor['namespace']].append(interface)
+            upstream_port_ids.append(port_id)
+
+    return ptf_src_port, upstream_port_ids, dut_port
+
+
+@pytest.fixture(name="setUp", scope="module")
+def fixture_setUp_vxlan_vnet_routes(duthosts,
+                                    rand_one_dut_hostname,
+                                    tbinfo,
+                                    ptfadapter,
+                                    prepare_test_port,
+                                    encap_type):
+    '''
+        Setup for the entire script.
+        The basic steps in VxLAN configs are:
+            1. Configure VxLAN tunnel.
+            2. Configure Vnet and its VNI.
+
+            The testcases are focused on the "configure routes" step. They add,
+            delete, modify, the routes.
+    '''
+    data = {}
+
+    ptf_src_port, ptf_dst_ports, dut_port = prepare_test_port
+    data['ptfadapter'] = ptfadapter
+    data['ptf_src_port'] = ptf_src_port
+    data['ptf_dst_ports'] = ptf_dst_ports
+    data['dut_port'] = dut_port
+
+    duthost = duthosts[rand_one_dut_hostname]
+    minigraph_data = duthost.get_extended_minigraph_facts(tbinfo)
+    data['minigraph_facts'] = minigraph_data
+    data['tbinfo'] = tbinfo
+    data['duthost'] = duthost
+    if data['minigraph_facts']['minigraph_lo_interfaces'][0]['prefixlen'] == 32:
```

Comment:
Better not to check the prefix length. You may check the address type to determine whether it's IPv4 or IPv6.